servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ClientParentChannelInitializer.java

```diff
@@ -84,4 +87,9 @@ public void channelRegistered(final ChannelHandlerContext ctx) {
             // Http2ConnectionHandler.processGoAwayWriteResult will close the connection after GO_AWAY is flushed
         }
     }
+
+    private static Http2Settings applyClientDefaultSettings(Http2Settings settings) {
+        // Notify server that this client does not support server push and request it to be disabled.
+        return settings.pushEnabled(false).maxConcurrentStreams(0L);
```

Comment:
This is hard to debug if we silently override these 2 settings. With a custom `Http2Settings` interface we can prevent users from specifying `pushEnabled` at the API level. For `maxConcurrentStreams`, consider validating settings at the client builder and throw an exception if users try to provide `maxConcurrentStreams`.